File: python/examples/non_unique_pii_markers.py
# Example script to illustrate how to make API calls to the Private AI API
# to deidentify text using the non-unique PII markers feature.

import dotenv
import os
from privateai_client import PAIClient, request_objects

# Use to load the API KEY for authentication
dotenv.load_dotenv()

# Check if the API_KEY and PAI_URL environment variables are set
if "API_KEY" not in os.environ:
    raise KeyError("API_KEY must be defined in .env to run the examples.")
if "PAI_URL" not in os.environ:
    raise KeyError("PAI_URL must be defined in .env to run the examples.")

# Client initialization
client = PAIClient(url=os.environ["PAI_URL"], api_key=os.environ["API_KEY"])

sample_text = "Hello, my name is May. I am the aunt of Pieter Parker. We live in Toronto, Canada."
sample_entity_detection = request_objects.entity_detection_obj(return_entity=True)
sample_processed_text = request_objects.processed_text_obj(
    type="MARKER",
    pattern="[BEST_ENTITY_TYPE]"
)

process_text_request = request_objects.process_text_obj(
    text=[sample_text],
    link_batch=False,
    entity_detection=sample_entity_detection,
    processed_text=sample_processed_text
)

response = client.process_text(process_text_request)
print(response.processed_text)

File: js/examples/non_unique_pii_markers.js
/**
 * Example script to illustrate how to make requests to the Private AI API
 * to deidentify text using the non-unique PII markers feature.
 */
const axios = require("axios");
const dotenv = require("dotenv");

// Use to load the API_KEY and URL
dotenv.config();

// Example without async/await
function sync_non_unique_pii_markers() {
  console.log("***** Sync non-unique PII markers *****");
  axios.post(
    `${process.env.PAI_URL}/process/text`,
    {
      text: ["My name is John and my friend is Grace"],
      link_batch: false,
      entity_detection: {
        return_entity: true,
      },
      processed_text: {
        type: "MARKER",
        pattern: "[BEST_ENTITY_TYPE]",
      },
    },
    {
      headers: {
        "Content-Type": "application/json",
        "X-API-KEY": process.env.API_KEY,
      },
    }
  )
  .then((result) => console.log(JSON.stringify(result.data, undefined, 2)))
  .catch((error) =>
    console.error(
      `The request failed with the status code ${error.response.status}`
    )
  );
}

// Example with async/await
async function async_non_unique_pii_markers() {
  console.log("***** Async non-unique PII markers *****");
  try {
    const result = await axios.post(
      `${process.env.PAI_URL}/process/text`,
      {
        text: ["My name is John and my friend is Grace"],
        link_batch: false,
        entity_detection: {
          return_entity: true,
        },
        processed_text: {
          type: "MARKER",
          pattern: "[BEST_ENTITY_TYPE]",
        },
      },
      {
        headers: {
          "Content-Type": "application/json",
          "X-API-KEY": process.env.API_KEY,
        },
      }
    );

    const { data } = result;

    console.log(JSON.stringify(data, undefined, 2));
  } catch (error) {
    console.error(
      `The request failed with the status code ${error.response.status}`
    );
  }
};

// sync_non_unique_pii_markers();
async_non_unique_pii_markers();
